src/cbc_sdk/enterprise_edr/threat_intelligence.py

```diff
@@ -482,14 +960,166 @@ def __init__(self, cb, model_unique_id=None, initial_data=None,
             self._iocs = IOC(cb, initial_data=self.iocs, report_id=self.id)
         if self.iocs_v2:
             self._iocs_v2 = [IOC_V2(cb, initial_data=ioc, report_id=self.id) for ioc in self.iocs_v2]
+        self._iocs_v2_need_sync = False
+
+    class ReportBuilder:
+        """Helper class allowing Reports to be assembled."""
+        def __init__(self, cb, report_body):
+            """
+            Initialize a new ReportBuilder.
+
+            Args:
+                cb (CBCloudAPI): A reference to the CBCloudAPI object.
+                report_body (dict): Partial report body which should be filled in with all "required" fields.
+            """
+            self._cb = cb
+            self._report_body = report_body
+            self._report_body['tags'] = []
```

Comment:
What if report_body contains tags? This is going to overwrite the key

Reply:
report_body will never contain tags because it's constructed in Report.create() which takes discrete parameters and builds the dict itself.

Reply:
You assume that is the only place some will call this when you can create a builder directly if desired

Reply:
Why would anyone create a builder directly? The create() function encapsulates that process.  In the alternative, if they already have a fully-formatted dict, they can construct a Report directly.

Nonetheless, I put in a guard clause that will only initialize the tags if they aren't already there (or are there but aren't a list).

Reply:
Thanks! I feel more comfortable with this implementation